### DISCUSSION TITLE
Firebase Adapter to use Admin SDK

### DISCUSSION THREAD

**Discussion by nhuethmayr** [nextauthjs/next-auth#5408] [Help]
Hi,

I'm using your `firebase-adapter` package and wonder why it is not using the **Firebase Admin SDK**? As I understand it, accessing the DB is performed inside the `/auth/[...nextAuth].ts` which runs server-side.

Using the client SDK in combination with next-auth means that I have to essentially have all my firestore database public.

![image](https://user-images.githubusercontent.com/26460692/191984912-3d204b29-4f2f-4b37-8f92-e7b659aa3f4b.png)

I did some research online and found that others also run into the same problem.
https://stackoverflow.com/questions/73261414/does-next-auth-really-only-works-without-firebase-security-rules
https://stackoverflow.com/questions/71654212/firebase-does-not-authenticate-when-using-next-auth?noredirect=1&lq=1
https://github.com/lowfront/next-auth-firestore-adapter-example

The proposed solution, to use a custom authentication token, seems needlessly complex and makes me worry about compatibility in the future.

**Comment by nhuethmayr** [comment]
I also found this https://github.com/nextauthjs/next-auth/issues/5049

**Comment by drunkenvalley** [comment]
I'm also a bit puzzled by the lack of a Firebase Admin SDK adapter. In the end I wrote one for myself using the Firebase Adapter here as a basis. I'm intending to try and contribute it to the project, though I want to do some more testing upfront and figure my way out through the guidelines before submitting it.

**Comment by nhuethmayr** [comment]
I have my own version of the adapter already - even submitted a PR. Nobody responded though and so I withdrew it.
If you have a version that is working, we could both contribute.

**Comment by drunkenvalley** [comment]
I went over the commit history, and something I notice is that they did previously use the firebase admin sdk. However, they changed from that to regular firestore. I'm not sure why yet, as the referenced pull request does not exist.

See pull request here referencing some issues that were discovered. https://github.com/nextauthjs/adapters/pull/108

Unfortunately, what those issues were is not mentioned, and the pull request in question is not available that I can see.

**Comment by nhuethmayr** [comment]
@balazsorban44 Can you elaborate on why you guys changed from `firebase-admin` to `firebase`?

**Comment by drunkenvalley** [comment]
I am curious though, @NorbertHuethmayr  - how did you solve the Typescript complaints ala these?

```typescript
Argument of type 'Omit<AdapterUser, "id">' is not assignable to parameter of type 'WithFieldValue<AdapterUser>'.
  Type 'Omit<AdapterUser, "id">' is missing the following properties from type '{ [x: string]: unknown; id: string | FieldValue; emailVerified: FieldValue | WithFieldValue<Date>; name?: string | FieldValue; email?: string | FieldValue; image?: string | FieldValue; }': id, emailVerifiedts(2345)
```

The code works, but it complains about it. For now I threw a `//@ts-ignore` on it, but I feel like I'm probably doing it wrong.

**Comment by nhuethmayr** [comment]
I did cast ‘as any’ - as I couldn’t find a way to solve it either
________________________________
